dev-tools/mage/dockerbuilder.go

```diff
@@ -66,25 +72,32 @@ func (b *dockerBuilder) Build() error {
 		return err
 	}
 
-	if err := b.prepareBuild(); err != nil {
-		return errors.Wrap(err, "failed to prepare build")
-	}
+	for _, variant := range variants {
+		// There is no UBI editition of the offline agent, so we skip this
+		if variant == "offline" && regexp.MustCompile("-ubi\\d+$").MatchString(b.imageName) {
+			continue
+		}
 
-	tries := 3
-	tag, err := b.dockerBuild()
-	for err != nil && tries != 0 {
-		fmt.Println(">> Building docker images again (after 10 seconds)")
-		// This sleep is to avoid hitting the docker build issues when resources are not available.
-		time.Sleep(10)
-		tag, err = b.dockerBuild()
-		tries -= 1
-	}
-	if err != nil {
-		return errors.Wrap(err, "failed to build docker")
-	}
+		if err := b.prepareBuild(variant); err != nil {
+			return errors.Wrap(err, "failed to prepare build")
+		}
 
-	if err := b.dockerSave(tag); err != nil {
-		return errors.Wrap(err, "failed to save docker as artifact")
+		tag, err := b.dockerBuild(variant)
+		tries := 3
+		for err != nil && tries != 0 {
+			fmt.Println(">> Building docker images again (after 10 seconds)")
+			// This sleep is to avoid hitting the docker build issues when resources are not available.
+			time.Sleep(10)
```

Comment:
This is 10ns not 10s as the comment indicates. (this already existed)

Reply:
I changed the timing to 10s, is that what you thought was more correct?